lib/ansible/cli/__init__.py

```diff
@@ -396,7 +396,8 @@ def validate_conflicts(self, vault_opts=False, runas_opts=False, fork_opts=False
 
     @staticmethod
     def unfrack_paths(option, opt, value, parser):
-        paths = getattr(parser.values, option.dest)
+        # If this callback is executed, we've been given explicit values, ignore the default value
+        paths = [v for v in getattr(parser.values, option.dest) if v not in option.default]
```

Comment:
Why not just not set a 'default' for the Option() if the goal is to ignore the default value?  Or set the default to an empty container as unfrack_paths intends. Then the callback doesn't need to care about the default

Reply:
My concern are the number of arguments throughout the CLI that use this callback already.  It's much easier to make the change in 1 location, instead of changing how we handle defaults throughout the code.  There is a much larger chance of introducing a bug if we go that route.

Although I agree, generally speaking, that would probably be preferable.